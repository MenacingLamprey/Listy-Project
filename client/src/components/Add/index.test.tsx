import { render, screen } from "@testing-library/react"; 
import userEvent from '@testing-library/user-event'
import { Add } from ".";

describe("<LoginForm />", () => {
  test("should be the same as snapshot", async () => {
    const { container } = render(<Add/>)
    expect(container.firstChild).toMatchSnapshot();
  });

  test("it renders a popup when button is clicked" ,() =>{
    const { container } = render(<Add/>)
    const popBtn = screen.getByTestId('addPop')

    let nullPostBtn = screen.queryByText(/POST/)
    let nullClosebtn = screen.queryByRole('button', {name :'CLOSE'})
    expect(nullPostBtn).toBeNull() 
    expect(nullClosebtn).toBeNull();

    userEvent.click(popBtn) //creates popup

    const postBtn = screen.getByText(/POST/)    // elements that exist on popup
    const closebtn = screen.getByRole('button', {name :'CLOSE'})
    screen.getByText(/add_photo_alternate/)

    userEvent.click(closebtn) //removes items on popup
    nullPostBtn = screen.queryByText(/POST/)
    nullClosebtn = screen.queryByRole('button', {name :'CLOSE'})
    expect(nullPostBtn).toBeNull() 
    expect(nullClosebtn).toBeNull();
    
    userEvent.click(popBtn) //creates popup

    const content = screen.getByPlaceholderText('movie/tv-show name') //get input fields
    const rating = screen.getByPlaceholderText('rating')
    const genre = screen.getByPlaceholderText('genre')

    userEvent.type(content, 'Interstellar');
    userEvent.type(rating,'5')
    userEvent.type(genre, 'sci-fi')

    userEvent.click(postBtn)
  })
});