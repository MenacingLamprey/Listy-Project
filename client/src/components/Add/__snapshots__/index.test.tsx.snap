// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should be the same as snapshot 1`] = `
<div>
  <link
    href="https://fonts.googleapis.com/icon?family=Material+Icons"
    rel="stylesheet"
  />
  <div
    class="addPopup"
    data-testid="addPop"
  >
    <img
      class="add-logo"
      src="https://icons.veryicon.com/png/o/object/material-design-icons/add-49.png"
    />
  </div>
  <section
    class="posts-container"
  />
</div>
`;
